src/utils/EventHandlers.tsx

```diff
@@ -79,3 +80,38 @@ export const handleMarkerOnClick = (
 ) => (): void => {
   dispatch({ type: "setCurrentCombinedRow", value: givenRow });
 };
+
+////////////////
+// Sidebar.tsx
+////////////////
+
+export const handleVisibilityToggle = (
+  stateManager: Types.stateManager
+) => (): void => {
+  const [state, dispatch] = stateManager;
+  const isSearchBarToggled = state.useSearchBar;
+  isSearchBarToggled &&
```

Comment:
Are you using short circuit evaluation as an abbreviated if? If yes: please don't. If no: what is this doing?

Reply:
Yes, in the case of handleVisibilityToggle I'd want the search bar filter to be disabled if the visibility filter is on and in the case of handleSearchBarToggle, I'd want the visibility filter to be disabled if the search bar filter is on. 

Is the short circuit a bad practice?

Reply:
I think so, yes. The nice thing about if statements is that they're so ubiquitous that everyone instantly understands what you're trying to express. Short circuit evaluation is also very common, but it's easy to forget that `&&` is a short circuited and and not just `and`. 
Believe it or not, there is a pretty common group of bugs related to this. If you have a block of code like so
```java
if (foo && funcThatReturnsBoolAndHasSideEffect(bar)) {
...
}
```
Often times people hit really confusing bugs because they forget that the side effect of `funcThatReturnsBoolAndHasSideEffect` only occurs when `foo` is `true`. Now, of course, there are other issues with this, namely that you have a function with a side effect in the predicate of your if. That said, it goes to show that it is very easy to forget that `&&` is not _just_ an `and`.